src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala

```diff
@@ -308,22 +308,51 @@ trait KorbHandler extends KorbStatusHandler
     }
   }
 
-  def modifyKoerbeForAboVertriebChange(abo: Abo, orig: Option[Abo])(implicit personId: PersonId, session: DBSession, publisher: EventPublisher): Unit = {
+  def modifyKoerbeForAboVertriebChange(abo: HauptAbo, orig: Option[HauptAbo])(implicit personId: PersonId, session: DBSession, publisher: EventPublisher): Unit = {
     logger.debug(s"modifyKoerbeForAboVertriebChange abo: $abo orig: $orig")
     for {
       originalAbo <- orig
       if (abo.vertriebId != originalAbo.vertriebId)
       abotyp <- stammdatenWriteRepository.getAbotypById(abo.abotypId)
     } yield {
-      stammdatenWriteRepository.getLieferungenOffenByAbotyp(abo.abotypId) map { lieferung =>
+      /* create or delete basket for the main abo*/
+      val offenLieferungenForMainAbo = stammdatenWriteRepository.getLieferungenOffenByAbotyp(abo.abotypId)
+      offenLieferungenForMainAbo map { lieferung =>
         if (lieferung.vertriebId == originalAbo.vertriebId) {
           deleteKorb(lieferung, originalAbo)
         }
         if (lieferung.vertriebId == abo.vertriebId) {
           upsertKorb(lieferung, abo, abotyp)
         }
-        recalculateNumbersLieferung(lieferung)
```

Comment:
recalculateNumbersLieferung() has been removed, but is included in the part for the Zusatzabo. Have you done this by purpose?

Reply:
recalculateNumbersLieferung for all Main Abo's is done at Line 355.